in_toto/verifylib.py

```diff
@@ -1423,6 +1471,27 @@ def in_toto_verify(layout, layout_key_dict, link_dir_path=".",
     step_name (optional): A name assigned to the returned link. This is mostly
         useful during recursive sublayout verification.
 
+    prerecorded_inspections (optional):
+        A dictionary of already performed inspections. Inspections present here
+        will not be executed and provided results will be recorded instead.
+        The dictionary format is as follows:
+
+        {
+            <inspection name> : {
+                "materials": {
+                    <material 1 filename>: {
+                        "sha256": <material 1 sha256 hash>,
+                        ...
+                    },
+                "products": {
+                    <product 1 filename>: {
+                        "sha256": <product 1 sha256 hash>,
+                        ...
+                    },
+                },
+            ...
+        }
+
```

Comment:
Please reformat this [as sphinx literal block](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#literal-blocks) so that it renders nicely on readthedocs ...

```suggestion
        The dictionary format is as follows::

            {
                <inspection name> : {
                    "materials": {
                        <material 1 filename>: {
                            "sha256": <material 1 sha256 hash>,
                            ...
                        },
                    "products": {
                        <product 1 filename>: {
                            "sha256": <product 1 sha256 hash>,
                            ...
                        },
                    },
                ...
            }

```

Reply:
Also, what if we just make `prerecorded_inspections` a dictionary of Metablock link objects? This would allow users to actually pass full prerecorded inspections, instead of only the materials/products aspects (maybe there is a use case for that?), and we could use existing Metablock/Link validate methods to check the inputs, and it would reduce your patch in  `run_all_inspections` to something like...

```python
    if inspection.name in prerecorded_inspections:
      LOG.info("Using pre-recorded inspection '{}' results...".format(
          inspection.name))

      inspection_links_dict[inspection.name] = \
          prerecorded_inspections[inspection.name]
      continue
```

Moreover we could then easily enhance the `in-toto-verify` cli to read prerecorded inspection link files. :)

I admit that it makes it slightly more difficult for the caller. But I think the `in_toto.models.*` provide enough tools to easily generate the corresponding objects.